src/postprocessing/EffectComposer.ts

```diff
@@ -1,103 +1,105 @@
-import {
-  Clock,
-  LinearFilter,
-  Mesh,
-  OrthographicCamera,
-  PlaneGeometry,
-  RGBAFormat,
-  Vector2,
-  WebGLRenderTarget,
-} from 'three'
+import { Clock, LinearFilter, RGBAFormat, Vector2, WebGLRenderer, WebGLRenderTarget } from 'three'
 import { CopyShader } from '../shaders/CopyShader'
-import { ShaderPass } from '../postprocessing/ShaderPass'
-import { MaskPass } from '../postprocessing/MaskPass'
-import { ClearMaskPass } from '../postprocessing/MaskPass'
-
-var EffectComposer = function (renderer, renderTarget) {
-  this.renderer = renderer
-
-  if (renderTarget === undefined) {
-    var parameters = {
-      minFilter: LinearFilter,
-      magFilter: LinearFilter,
-      format: RGBAFormat,
-    }
+import { ShaderPass } from './ShaderPass'
+import { MaskPass, ClearMaskPass } from './MaskPass'
+import { Pass } from './Pass'
+
+class EffectComposer {
+  renderer: WebGLRenderer
+  private _pixelRatio: number
+  private _width: number
+  private _height: number
+  renderTarget1: WebGLRenderTarget
+  renderTarget2: WebGLRenderTarget
+  writeBuffer: WebGLRenderTarget
+  readBuffer: WebGLRenderTarget
+  renderToScreen: boolean
+  passes: Pass[] = []
+  copyPass: Pass
+  clock: Clock
+
+  constructor(renderer: WebGLRenderer, renderTarget: WebGLRenderTarget) {
+    this.renderer = renderer
 
-    var size = renderer.getSize(new Vector2())
-    this._pixelRatio = renderer.getPixelRatio()
-    this._width = size.width
-    this._height = size.height
-
-    renderTarget = new WebGLRenderTarget(this._width * this._pixelRatio, this._height * this._pixelRatio, parameters)
-    renderTarget.texture.name = 'EffectComposer.rt1'
-  } else {
-    this._pixelRatio = 1
-    this._width = renderTarget.width
-    this._height = renderTarget.height
-  }
+    if (renderTarget === undefined) {
+      var parameters = {
```

Comment:
`const` or `let`, never `var` please.

Reply:
Agree. Maybe we should add this as an eslint rule for all TypeScript files? Intentionally leaving out the JS files and make sure people conform to that style for new stuff?